serveradmin/views.py

```diff
@@ -42,3 +46,54 @@ def power(request):
             call(["sudo", "reboot"])
 
     return render(request, 'serveradmin/power.html')
+
+
+@staff_member_required
+def backup(request):
+    if request.POST:
+        if 'do_backup' in request.POST:
+            backup = Backup.create()
+            msg = _('Succesfully created backup {filename}').format(
+                filename=backup.name
+            )
+            messages.add_message(request, messages.SUCCESS, msg)
+        elif 'do_upload' in request.POST:
+            if 'upload' in request.FILES:
+                file_ = request.FILES['upload']
+                try:
+                    backup = Backup.load(file_)
+                except:
+                    messages.add_message(request, messages.ERROR,
+                                         _('Unable to load file.'))
+                else:
+                    msg = _('File {name} has been loaded.').format(
+                        name=backup.name)
+                    messages.add_message(request, messages.SUCCESS, msg)
+            else:
+                messages.add_message(request, messages.ERROR,
+                                     _('No file found to upload.'))
+        elif 'backup' in request.POST:
```

Comment:
I'm always a bit scarred when not using Django's forms and doing that kind of things `return os.remove(self.path)` without any check ;)